main.js

```diff
@@ -0,0 +1,256 @@
+"use strict";
+
+const {context} = require("@arangodb/locals");
+const createRouter = require("@arangodb/foxx/router");
+const {query, db} = require("@arangodb");
+const graph_module = require("@arangodb/general-graph");
+const queues = require("@arangodb/foxx/queues");
+const joi = require("joi");
+const {modelTypes, metadataCollectionName} = require("./scripts/setup");
+
+const router = createRouter();
+context.use(router);
+
+// TODO: Make this model specific?
+const batch_size = 64;
+exports.BATCH_SIZE = batch_size;
+
+const embeddingQueueName = "embeddings_generation";
+
+function retrieveModel(modelName, modelType) {
+    const metadata_col = db._collection(metadataCollectionName);
+    const model_info = query`
+        FOR m in ${metadata_col}
+        FILTER m.name == ${modelName}
+        AND m.model_type == ${modelType}
+        RETURN m
+    `.toArray()
+
+    if (model_info.length > 0) {
+        return model_info[0];
+    }
+    // if we don't have a model, return null
+    return null;
+}
+
+function sendInvalidInputMessage(res, message) {
+    res.sendStatus(422);
+    res.json(message);
```

Comment:
You can use `res.throw` to make this a bit more explicit and avoid having to manually check return codes (it throws an exception Foxx recognizes): https://www.arangodb.com/docs/stable/foxx-reference-routers-response.html#throw